### PR TITLE
Install external dependencies at runtime

### PR DESCRIPTION
**Pull request by XinxingLiu**
To demonstrating that we can install external dependencies (including Z3, solc, Corral and Boogie) at VeriSol runtime. Run as expected on Windows. 

Tasks already done:
* Create VeriSol.nupkg as dotnet tool package
* Test the package installation scenario
* Update azure pipeline to run VeriSol.exe and SolToBoogieTest.exe
* Clean up code, for example, remove git submodule(s), removing directories like Binaries, Tool, Sources\VeriSolRunner
* Update documentations

ToDos:
* Manage runtime generated files, like corral.txt
* Test and fix bugs on mac and Linux
* Publish VeriSol to nuget.org

### CODE REVIEW

**Comment by shuvendu-lahiri** [comment]
Let us also try to have the build pipeline for ubuntu at least. As it is critical for non-windows users.

**Comment by shuvendu-lahiri** [comment]
To run the regression script with source installation, do we need to install the package? Or is this only if you don't want the sources.

**Comment by shuvendu-lahiri** [comment]
I think we should not expose the nuget install (in the INSTALL.md) until we have published the nuget package (I want to do that after a few more features/bug fixes in two weekes). We can wait for it or try to merge the PR without instructions for nuget for users.

**Comment by XinxingLiu** [comment]
> I think we should not expose the nuget install (in the INSTALL.md) until we have published the nuget package (I want to do that after a few more features/bug fixes in two weekes). We can wait for it or try to merge the PR without instructions for nuget for users.

No it doesn't matter if we published the nuget package yet. In the updated INSTALL.md we mentioned the package need to be installed from local path by specifying local nuget feed with
`--add-source %VERISOL_PATH%\nupkg\` in the install command.

**Comment by XinxingLiu** [comment]
> To run the regression script with source installation, do we need to install the package? Or is this only if you don't want the sources.

Currently there are two nuget cli tools: VeriSol and SoltoBoogieTest. To run the regression tests, you need (and only need) the second one installed.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -6,46 +6,36 @@
...
+
+   VeriSol $(Build.SourcesDirectory)/Test/regressions/ERC20-simplified.sol ERC20 /tryProof /tryRefutation:10 /printTransactionSequence
+   type boogie.txt
   displayName: 'Run VeriSol on some examples'
-- script: | 
-     dotnet $(Build.SourcesDirectory)/Binaries/SolToBoogieTest.dll $(Build.SourcesDirectory)/Test/
-  displayName: 'Run regression tests'
+
+- task: DotNetCoreCLI@2
+  displayName: 'Install SolToBoogieTest as dotnet cli tool'
```
Comment (shuvendu-lahiri) [inline azure-pipelines.yml]: I don't think we publish VeriSolregression runner as a dotnet tool on nuget. Do you think this is only for developers?

**Pull request closed without merging**